src/grammar/grammar.ohm

```diff
@@ -20,6 +20,9 @@ Tact {
     PrimitiveTypeDecl = primitive typeId ";"
 
     ModuleFunction = FunctionDefinition
+                   | AsmFunction
+
+    AsmFunction = "asm" AsmShuffle? FunctionAttribute* fun id Parameters (":" Type)? "{" asmInstruction+ "}"
```

Comment:
Great! Now, to also support blocked instructions I'm thinking of something like this:

```ohm
AsmFunction = "asm" AsmShuffle? FunctionAttribute* fun id Parameters (":" Type)? "{" AsmInstruction+ "}"

// ...

AsmInstruction = AsmInstructionBlock | asmInstruction
AsmInstructionBlock = "{" AsmInstruction+ "}"

// ...

asmInstruction = ("x" | "X" | "b" | "B") "{" hexDigit* "}" --data
               | ~comment (~space any)+ --any
```

Reply:
Getting all the grammar right as a non-goal of this PR: we need a solution which would allow us to port `asm` functions from `stdlib.fc` and `mathlib.fc`.